temporalio/client.py

```diff
@@ -595,6 +603,44 @@ def get_workflow_handle_for(
             result_type=defn.ret_type,
         )
 
+    async def list_workflows(
+        self,
+        query: Optional[str] = None,
+        *,
+        page_size: int = 1000,
+        next_page_token: Optional[bytes] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowExecutionAsyncIterator:
+        """List workflows.
+
+        This eagerly requests the first page and will raise an exception if
+        not available.
```

Comment:
Hmm.. why?

I think most users would rather use this with `async for client.list_workflows(...)`.
This API is a bit weird to me.

Reply:
Yeah, I wanted this to error for something like invalid query string before they passed around the iterator to something else. But I can just make it wait to error.